src/api/client.ts

```diff
@@ -1,4 +1,37 @@
-import createClient from 'openapi-fetch';
-import type { paths } from './schema';
+import { setCsrfToken } from '@/state/csrf/csrfSlice';
+import { store } from '@/state/store';
+import createClient, { Middleware } from 'openapi-fetch';
+import type { components, paths } from './schema';
 
 export const client = createClient<paths>();
+
+const csrfMiddleware: Middleware = {
+  async onRequest(request) {
+    const state = store.getState();
+    let csrf = state.root?.csrf?.token;
+    if (!csrf) {
+      const response = await fetch('/api/csrf', {
+        signal: request.signal,
+      });
+      if (response.ok) {
+        csrf = (await response.json()) as components['schemas']['CsrfTokenDto'];
+        store.dispatch(setCsrfToken(csrf));
+      }
+    }
```

Comment:
Maybe the middleware should not try to fetch the csrf token; it is only required when the user is logged in, and parts of the app do not require authentication. Also, "safe" requests (get, options, head) do not require the csrf token in any case.

Reply:
Fetching removed from middleware, it's now only fetched on login 👍